handlers/external/external.go

```diff
@@ -344,13 +356,41 @@ func (h *handler) SendMsg(ctx context.Context, msg courier.Msg) (courier.MsgStat
 	return status, nil
 }
 
+func buildQuickRepliesResponse(quickReplies []string, sendMethod string, contentType string) string {
+	var response, format, separator string
+
+	switch {
+	case (sendMethod == http.MethodPost || sendMethod == http.MethodPut) && contentType == contentJSON:
+		response = "[]"
+		format = "[\"%s\"]"
+		separator = "\",\""
+
+	case (sendMethod == http.MethodPost || sendMethod == http.MethodPut) && contentType == contentXML:
+		response = ""
+		format = "<item>%s</item>"
+		separator = "</item><item>"
+
+	default:
+		response = ""
+		format = "%s"
+		separator = "\\,"
+	}
+
+	if len(quickReplies) > 0 {
+		response = fmt.Sprintf(format, strings.Join(quickReplies, separator))
+	}
+	return response
+}
+
 func replaceVariables(text string, variables map[string]string, contentType string) string {
 	for k, v := range variables {
 		// encode according to our content type
 		switch contentType {
 		case contentJSON:
-			marshalled, _ := json.Marshal(v)
-			v = string(marshalled)
+			if !(len(v) > 1 && v[0] == '[' && v[len(v) - 1] == ']') {
```

Comment:
This is a bit hacky and almost certainly going to blow up at some point. Better would be to pass in whether the thing being passed in needs to be escaped or not and skip over this in the case where it isn't. (IE: escaping needs to take place in `buildQuickReplies` then that value is passed to this with escaping off)